integr8ly/grafana-operator
# Import datasources

Fixes #75 

Adds an optional `datasources` section to dashboards that allows to map existing datasources to defined inputs in the dashboard. Provides an example for this using keycloak dashboard.

The idea is that an input section in an exported dashboard of:

```
      "__inputs": [
        {
          "name": "DS_PROMETHEUS",
          "label": "Prometheus",
          "description": "",
          "type": "datasource",
          "pluginId": "prometheus",
          "pluginName": "Prometheus"
        }
      ],
```

can be mapped to an existing datasource with the name `Prometheus` using:

```
...
spec:
  datasources:
    - inputName: "DS_PROMETHEUS"
      datasourceName: "Prometheus"
...
```

Verification steps:

1. Deploy Grafana and create the Prometheus example datasource
2. Create the KeycloakDashboard.yaml example
3. Open the dashboard and make sure it correctly identifies the datasource (there may still be errors if the datasource fails to find a prometheus instance)
